test/realm_set_test.dart

```diff
@@ -0,0 +1,535 @@
+      case RealmObject:
+        return Sets(objectsSet as RealmSet<Car>, [Car("Tesla"), Car("VW"), Car("Audi")], (realm, value) => realm.find<Car>((value as Car).make));
+      case _NullableBool:
+        return Sets(nullableBoolSet as RealmSet<bool?>, [...setByType(bool).values, null]);
+      case _NullableInt:
+        return Sets(nullableIntSet as RealmSet<int?>, [...setByType(int).values, null]);
+      case _NullableString:
+        return Sets(nullableStringSet as RealmSet<String?>, [...setByType(String).values, null]);
+      case _NullableDouble:
+        return Sets(nullableDoubleSet as RealmSet<double?>, [...setByType(double).values, null]);
+      case _NullableDateTime:
+        return Sets(nullableDateTimeSet as RealmSet<DateTime?>, [...setByType(DateTime).values, null]);
+      case _NullableObjectId:
+        return Sets(nullableObjectIdSet as RealmSet<ObjectId?>, [...setByType(ObjectId).values, null]);
+      case _NullableUuid:
+        return Sets(nullableUuidSet as RealmSet<Uuid?>, [...setByType(Uuid).values, null]);
+      default:
+        throw RealmError("Unsupported type $type");
+    }
+  }
+
+  List<Object?> values(Type type) {
+    return setByType(type).values;
+  }
+
+  List<Object?> getValuesOrManagedValues(Realm realm, Type type) {
+    Sets set = setByType(type);
+    if (set.values.first is! RealmObject) {
+      return set.values;
+    }
+
+    return set.values.map<Object?>((value) {
+      RealmObject? realmValue = set.getRealmObject!(realm, value as RealmObject);
+      return realmValue;
+    }).toList();
+  }
+}
+
+class Sets {
+  final RealmSet<Object?> set;
+  final List<Object?> values;
+  RealmObject? Function(Realm realm, RealmObject value)? getRealmObject = (realm, value) => value;
+
+  Sets(this.set, this.values, [this.getRealmObject]);
+}
+
+Future<void> main([List<String>? args]) async {
+  await setupTests(args);
+
+  for (var type in supportedTypes) {
+    test('RealmSet<$type> unmanged set add', () {
+      final testSet = TestRealmSets(1);
+      final set = testSet.setByType(type).set;
+      final values = testSet.values(type);
+
+      set.add(values.first);
+      expect(set.length, equals(1));
+      expect(set.contains(values.first), true);
+
+      set.remove(values.first);
+      expect(set.length, equals(0));
+      expect(set.contains(values.first), false);
+    });
+
+    test('RealmSet<$type> unmanged set remove', () {
+      final testSet = TestRealmSets(1);
+      final set = testSet.setByType(type).set;
+      final values = testSet.values(type);
+
+      set.add(values.first);
+      expect(set.length, equals(1));
+      expect(set.contains(values.first), true);
+
+      set.remove(values.first);
+      expect(set.length, equals(0));
+      expect(set.contains(values.first), false);
+    });
+
+    test('RealmSet<$type> unmanged set elementAt', () {
```

Comment:
I guess it's confusing because `set` means both "a set of values" and "set a value". In this case I believe Lubo is using it as the "unmanaged set" noun. Perhaps it may be clearer if the test was renamed to something like `'RealmSet<$type>.elementAt (unmanaged)`.

Reply:
fixed